File: c++/examples/case1.cpp
/*
 * case1.cpp [version 1.0]
 * CoSniWa:  COde SNIppet stopWAtch - case 1
 *
 * case1:  Test usleep() function using Cosniwa
 *
 * read more on: www.speedupcode.com
 *
 * (c) Jacek Pierzchlewski, 2017  jacek@pierzchlewski.com
 * license: BSD-2-Clause.
 */
#include <unistd.h>
#include "cosniwa.h"


void test_usleep(int ius, int iRego, Cosniwa* csw)
{
/*
 *  test_usleep:  TEST 'usleep()' by calling it 1000 times
 *
 *  Parameters:
 *   ius:    [int]       Argument for usleep() function
 *   iRego:  [int]       Cosniwa registration for the current test
 *   csw:    [Cosniwa*]  Pointer to Cosniwa class
 *
 */

    // Call usleep() function 1000 times
    for (int inxCall=0; inxCall < 1000; inxCall++)
    {
        csw->call_start(iRego);
        usleep(ius);
        csw->call_stop(iRego);
    }
}


int main()
{
    // Create Cosniwa object
    Cosniwa csw = Cosniwa();

    // Test usleep with arguments 1 [us] - 9 [us]
    test_usleep(1, csw.reg_code("Test on usleep(1) [1us] "), &csw);
    test_usleep(2, csw.reg_code("Test on usleep(2) [2us] "), &csw);
    test_usleep(3, csw.reg_code("Test on usleep(3) [3us] "), &csw);
    test_usleep(4, csw.reg_code("Test on usleep(4) [4us] "), &csw);
    test_usleep(5, csw.reg_code("Test on usleep(5) [5us] "), &csw);
    test_usleep(6, csw.reg_code("Test on usleep(6) [6us] "), &csw);
    test_usleep(7, csw.reg_code("Test on usleep(7) [7us] "), &csw);
    test_usleep(8, csw.reg_code("Test on usleep(8) [8us] "), &csw);
    test_usleep(9, csw.reg_code("Test on usleep(9) [9us] "), &csw);

    // Test usleep with arguments  10 [us] - 90 [us]
    test_usleep(10, csw.reg_code("Test on usleep(10) [10us] "), &csw);
    test_usleep(20, csw.reg_code("Test on usleep(20) [20us] "), &csw);
    test_usleep(30, csw.reg_code("Test on usleep(30) [30us] "), &csw);
    test_usleep(40, csw.reg_code("Test on usleep(40) [40us] "), &csw);
    test_usleep(50, csw.reg_code("Test on usleep(50) [50us] "), &csw);
    test_usleep(60, csw.reg_code("Test on usleep(60) [60us] "), &csw);
    test_usleep(70, csw.reg_code("Test on usleep(70) [70us] "), &csw);
    test_usleep(80, csw.reg_code("Test on usleep(80) [80us] "), &csw);
    test_usleep(90, csw.reg_code("Test on usleep(90) [90us] "), &csw);

    // Test usleep with arguments  100 [us] - 900 [us]
    test_usleep(100, csw.reg_code("Test on usleep(100) [100us] "), &csw);
    test_usleep(200, csw.reg_code("Test on usleep(200) [200us] "), &csw);
    test_usleep(300, csw.reg_code("Test on usleep(300) [300us] "), &csw);
    test_usleep(400, csw.reg_code("Test on usleep(400) [400us] "), &csw);
    test_usleep(500, csw.reg_code("Test on usleep(500) [500us] "), &csw);
    test_usleep(600, csw.reg_code("Test on usleep(600) [600us] "), &csw);
    test_usleep(700, csw.reg_code("Test on usleep(700) [700us] "), &csw);
    test_usleep(800, csw.reg_code("Test on usleep(800) [800us] "), &csw);
    test_usleep(900, csw.reg_code("Test on usleep(900) [900us] "), &csw);

    // Test usleep with arguments  1 [ms] - 9 [ms]
    test_usleep(1000, csw.reg_code("Test on usleep(1000) [1ms] "), &csw);
    test_usleep(2000, csw.reg_code("Test on usleep(2000) [2ms] "), &csw);
    test_usleep(3000, csw.reg_code("Test on usleep(3000) [3ms] "), &csw);
    test_usleep(4000, csw.reg_code("Test on usleep(4000) [4ms] "), &csw);
    test_usleep(5000, csw.reg_code("Test on usleep(5000) [5ms] "), &csw);
    test_usleep(6000, csw.reg_code("Test on usleep(6000) [6ms] "), &csw);
    test_usleep(7000, csw.reg_code("Test on usleep(7000) [7ms] "), &csw);
    test_usleep(8000, csw.reg_code("Test on usleep(8000) [8ms] "), &csw);
    test_usleep(9000, csw.reg_code("Test on usleep(9000) [9ms] "), &csw);

    // Print out the timing results
    csw.resultc();
}

File: python/examples/case1.py
#
# case1.py [version 1.0]
# CoSniWa:  COde SNIppet stopWAtch [Python port] - case 1
#
# case1:  Test time.sleep() function using Cosniwa
#
# read more on: www.speedupcode.com
#
# (c) Jacek Pierzchlewski, 2017  jacek@pierzchlewski.com
# license: BSD-2-Clause.
#
try:
    import cCosniwa as csw
except ImportError:
    print("\nERROR: cCosniwa was not found! \n")
import time


def test_usleep(tSleep, iRego):

    """
    test_usleep:  TEST 'sleep()' by calling it 1000 times

    Parameters:
     ius:    [int]       Argument for sleep() function
     iRego:  [int]       Cosniwa registration for the current test
     csw:    [Cosniwa*]  Pointer to Cosniwa class
    """

    # Call sleep() function 1000 times
    for inxCall in range(1000):
        csw.call_start(iRego)
        time.sleep(tSleep)
        csw.call_stop(iRego)


def main():

    ius = 1e-6   # Definition of microsecond
    ims = 1e-6   # Definition of milisecons

    # Test usleep with arguments 1 [us] - 9 [us]
    test_usleep(1 * ius, csw.reg_code("Test on sleep(1us) "))
    test_usleep(2 * ius, csw.reg_code("Test on sleep(2us) "))
    test_usleep(3 * ius, csw.reg_code("Test on sleep(3us) "))
    test_usleep(4 * ius, csw.reg_code("Test on sleep(4us) "))
    test_usleep(5 * ius, csw.reg_code("Test on sleep(5us) "))
    test_usleep(6 * ius, csw.reg_code("Test on sleep(6us) "))
    test_usleep(7 * ius, csw.reg_code("Test on sleep(7us) "))
    test_usleep(8 * ius, csw.reg_code("Test on sleep(8us) "))
    test_usleep(9 * ius, csw.reg_code("Test on sleep(9us) "))

    # Test usleep with arguments  10 [us] - 90 [us]
    test_usleep(10 * ius, csw.reg_code("Test on sleep(10us) "))
    test_usleep(20 * ius, csw.reg_code("Test on sleep(20us) "))
    test_usleep(30 * ius, csw.reg_code("Test on sleep(30us) "))
    test_usleep(40 * ius, csw.reg_code("Test on sleep(40us) "))
    test_usleep(50 * ius, csw.reg_code("Test on sleep(50us) "))
    test_usleep(60 * ius, csw.reg_code("Test on sleep(60us) "))
    test_usleep(70 * ius, csw.reg_code("Test on sleep(70us) "))
    test_usleep(80 * ius, csw.reg_code("Test on sleep(80us) "))
    test_usleep(90 * ius, csw.reg_code("Test on sleep(90us) "))

    # Test usleep with arguments  100 [us] - 900 [us]
    test_usleep(100 * ius, csw.reg_code("Test on sleep(100us) "))
    test_usleep(200 * ius, csw.reg_code("Test on sleep(200us) "))
    test_usleep(300 * ius, csw.reg_code("Test on sleep(300us) "))
    test_usleep(400 * ius, csw.reg_code("Test on sleep(400us) "))
    test_usleep(500 * ius, csw.reg_code("Test on sleep(500us) "))
    test_usleep(600 * ius, csw.reg_code("Test on sleep(600us) "))
    test_usleep(700 * ius, csw.reg_code("Test on sleep(700us) "))
    test_usleep(800 * ius, csw.reg_code("Test on sleep(800us) "))
    test_usleep(900 * ius, csw.reg_code("Test on sleep(900us) "))

    # Test usleep with arguments  1 [ms] - 9 [ms]
    test_usleep(1 * ims, csw.reg_code("Test on sleep(1ms) "))
    test_usleep(2 * ims, csw.reg_code("Test on sleep(2ms) "))
    test_usleep(3 * ims, csw.reg_code("Test on sleep(3ms) "))
    test_usleep(4 * ims, csw.reg_code("Test on sleep(4ms) "))
    test_usleep(5 * ims, csw.reg_code("Test on sleep(5ms) "))
    test_usleep(6 * ims, csw.reg_code("Test on sleep(6ms) "))
    test_usleep(7 * ims, csw.reg_code("Test on sleep(7ms) "))
    test_usleep(8 * ims, csw.reg_code("Test on sleep(8ms) "))
    test_usleep(9 * ims, csw.reg_code("Test on sleep(9ms) "))

    # Print out the timing results
    csw.resultc()

if __name__ == '__main__':
    main()
